src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py

```diff
@@ -643,6 +643,19 @@ def test_provider_registration(self):
             self.assertTrue(result['registrationState'] in ['Registering', 'Registered'])
 
 
+class ProviderMGRegistrationTest(ScenarioTest):
+
+    def test_provider_mg_registration(self):
+
+        self.kwargs.update({'prov': 'Microsoft.ClassicInfrastructureMigrate', 'mg': self.create_random_name('azure-cli-management', 30)})
+
+        result = self.cmd('provider show -n {prov}').get_output_in_json()
+        if result['registrationState'] == 'Unregistered':
```

Comment:
please not use if in test scenarios.
just assert Unregistered status and then register the mg.